src/RadixDlt.NetworkGateway.Abstractions/CoreCommunications/TransactionSubmitter.cs

```diff
@@ -0,0 +1,240 @@
+
+    public record OtherSubmissionError(Exception Exception, string ErrorTitle) : NodeSubmissionResult;
+
+    private NodeSubmissionResult()
+    {
+    }
+
+    public bool ShouldStopSubmittingPermanently()
+    {
+        return this switch
+        {
+            IntentAlreadyCommitted => true,
+            AcceptedIntoMempool => false,
+            Rejection rejection => rejection.Details.IsPayloadRejectionPermanent,
+            MempoolPriorityThresholdNotMet => false,
+            _ => throw new ArgumentOutOfRangeException(),
+        };
+    }
+
+    public PendingTransactionPayloadLedgerStatus PayloadStatus()
+    {
+        return this switch
+        {
+            IntentAlreadyCommitted commit => commit.CommittedAsDetails.IsSameTransaction
+                ? PendingTransactionPayloadLedgerStatus.CommitPendingOutcomeUnknown
+                : PendingTransactionPayloadLedgerStatus.CommitOfOtherPayloadForIntentPendingOutcomeUnknown,
+            AcceptedIntoMempool => PendingTransactionPayloadLedgerStatus.TransientlyAccepted,
+            Rejection rejection => rejection.Details.IsPayloadRejectionPermanent
+                ? PendingTransactionPayloadLedgerStatus.PermanentlyRejected
+                : PendingTransactionPayloadLedgerStatus.TransientlyRejected,
+            _ => PendingTransactionPayloadLedgerStatus.Unknown,
+        };
+    }
+
+    public PendingTransactionIntentLedgerStatus IntentStatus()
+    {
+        return this switch
+        {
+            IntentAlreadyCommitted => PendingTransactionIntentLedgerStatus.CommitPendingOutcomeUnknown,
+            AcceptedIntoMempool => PendingTransactionIntentLedgerStatus.PossibleToCommit,
+            Rejection rejection => rejection.Details.IsIntentRejectionPermanent
+                ? PendingTransactionIntentLedgerStatus.PermanentRejection
+                : PendingTransactionIntentLedgerStatus.LikelyButNotCertainRejection,
+            _ => PendingTransactionIntentLedgerStatus.Unknown,
+        };
+    }
+
+    public string? GetExecutionFailureReason()
+    {
+        return this switch
+        {
+            Rejection rejection => rejection.Details.ErrorMessage,
+            _ => null,
+        };
+    }
+
+    public string? GetApiSubmitErrorMessage()
+    {
+        return this switch
+        {
+            MempoolPriorityThresholdNotMet { Details: var details } => $"Insufficient tip percentage of {details.TipPercentage}; min tip percentage {details.MinTipPercentageRequired}",
+            OtherSubmissionError details => details.ErrorTitle,
+            _ => null,
+        };
+    }
+}
+
+public interface ITransactionSubmitterObserver
+{
+    ValueTask ObserveSubmitAttempt(SubmitContext context);
+
+    ValueTask ObserveSubmitResult(SubmitContext context, NodeSubmissionResult nodeSubmissionResult);
+}
+
+public record SubmitContext(bool IsResubmission);
```

Comment:
Hmmm... shouldn't this context type also convey notarized TX blob and network id/name?

Reply:
Possibly 👍